### ISSUE TITLE
Is max_speed used in Meili Map Matching?

### ISSUE THREAD

**Issue by ward-vl** [valhalla/valhalla#3979]
Hi,

When timestamps are given together with the GPS trace, will Valhalla use this information to improve routing based on max_speed of segments in OSM?
If yes, in what way is this information used?

**Comment by kevinkreiser** [comment]
yes. it is used to cut off the routing computation before it could generate paths that would take significantly longer than the time the timestamps say it took. we have a configuration parameter that controls how much (by a multiplier) longer a found path can be in time, vs the actual measure time via timestamp. the default is quite generous though (5x if i remember right).

this is useful mostly for paths that would make big jumps in the graph for a given pair of edge candidates:

1. crossing carriageways, switching between one side of a motorway and the other
2. crossing levels (to use osm terminology), switch from a bridge to the road that goes under the bridge

basically anything that would require a very large amount of travel to get from one candidate edge to another in the graph.
